geomstats/geometry/hyperbolic_space.py

```diff
@@ -187,6 +275,170 @@ def extrinsic_to_intrinsic_coords(self, point_extrinsic):
 
         return point_intrinsic
 
+    @staticmethod
+    def _extrinsic_to_ball_coordinates(point):
+        """
+        Convert the parameterization of a point on the Hyperbolic space
+        from its intrinsic coordinates, to the poincare ball model
+        coordinates.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dimension + 1] in extrinsic
+                coordinates
+
+        Returns
+        -------
+        point_ball : array-like, shape=[n_samples, dimension] in
+                     poincare ball coordinates
+        """
+        return point[:, 1:] / (1 + point[:, :1])
+
+    @staticmethod
+    def _ball_to_extrinsic_coordinates(point):
+        """
+        Convert the parameterization of a point on the Hyperbolic space
+        from its poincare ball model coordinates, to the extrinsic
+        coordinates.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dimension] in Poincare ball
+                coordinates
+
+        Returns
+        -------
+        extrinsic : array-like, shape=[n_samples, dimension + 1] in
+                     extrinsic coordinate
+        """
+        squared_norm = gs.sum(point**2, -1)
+        denominator = 1-squared_norm
+        t = gs.to_ndarray((1+squared_norm)/denominator, to_ndim=2, axis=1)
+        expanded_denominator = gs.expand_dims(denominator, -1)
+        expanded_denominator = gs.repeat(expanded_denominator,
+                                         point.shape[-1], -1)
+        intrasic = (2*point)/expanded_denominator
+        return gs.concatenate([t, intrasic], -1)
+
+    @staticmethod
+    def _half_plane_to_extrinsic_coordinates(point):
+        """
+        Convert the parameterization of a point on the Hyperbolic space
+        from its upper half plane model coordinates, to the extrinsic
+        coordinates.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dimension] in Poincare ball
+                coordinates
+
+        Returns
+        -------
+        extrinsic : array-like, shape=[n_samples, dimension + 1] in
+                     extrinsic coordinate
+        """
+        assert(point.shape[-1] == 2)
+        x, y = point[:, 0], point[:, 1]
```

Comment:
avoid single letter for variable names, as it makes automatic replace impossible

Reply:
same on the lines below